mne/preprocessing/maxwell.py

```diff
@@ -0,0 +1,489 @@
+# Authors: Mark Wronkiewicz <wronk.mark@gmail.com>
+#          Jussi Nurminen <jnu@iki.fi>
+
+# This code was adapted and relicensed (with BSD form) with permission from
+# Jussi Nurminen
+
+# License: BSD (3-clause)
+
+# Note, there are an absurd number of different possible notations for
+# spherical coordinates, which confounds the notation for spherical harmonics.
+# Here, we purposefully stay away from shorthand notation in both and use
+# explicit terms (like 'azimuth' and 'polar') to avoid confusion.
+
+# TODO: write in equation numbers from Samu's paper
+
+import numpy as np
+from scipy.special import lpmv
+from scipy.linalg import pinv
+from ..forward._compute_forward import _concatenate_coils
+#from ..fixes import partial
+from scipy.misc import factorial as fact
+from ..io.constants import FIFF
+from .. import pick_types, pick_info
+from ..utils import logger
+from ..forward._make_forward import _read_coil_defs, _create_coils
+
+
+def maxwell_filter(raw, coils, origin, int_order=8, ext_order=3, n_jobs=1):
+    """Apply Maxwell filter to data using spherical harmonics.
+
+    Parameters
+    ----------
+    raw : instance of mne.io.Raw
+        Data to be filtered
+    coils : list
+        List of MEG coils. Each element must a contain coil information dict
+        continaining 'rmag', 'cosmag', and 'w' keys
```

Comment:
Interesting, so is one more correct than the other or are they just
different? I'd assume Elekta's would be better...

To answer the original question,  you should never have to deal with the
case where raw does not provide proper information to get parameters from
the coil definitions file.